assets/js/algolia-results.js

```diff
@@ -199,17 +199,29 @@ export class AlgoliaResults {
   }
 
   render(results) {
+    // We don't want to render pages where the URL is reference to a node.
+    // That means that the result is a page that has been unpublished.
+    const filtered_pages_hits = results.pages.hits.filter(
+      result => !/node\//.test(result._search_result_url)
```

Comment:
So this assumption is false. There have been and will be `search_result` type drupal results where the value of the `_search_result_url` field will look like this and be 100% valid:

`"entity:node/5019"`

This is due to the potential way editors may (and should) be entering these URLs in the UI, where they may use a Drupal autocomplete to search for the node they wish to connect to this item, click on it, and store the system path (which is always what we want for internal links, due to relative URLs being ephemeral). This lets Drupal resolve it to the current canonical path.